File: backend/app/models/moon_enrichment.py
"""Pydantic schemas for FreeAstro moon enrichment API."""

from typing import Literal

from pydantic import BaseModel, Field


class MoonEnrichmentEntry(BaseModel):
    date: str
    phase_name: str
    illumination_pct: float
    age_days: float | None = None
    is_waxing: bool | None = None
    special_labels: list[str] = Field(default_factory=list)
    visual_url: str | None = None


MoonEnrichmentStatus = Literal["complete", "partial", "pending", "unavailable"]


class MoonEnrichmentResponse(BaseModel):
    entries: list[MoonEnrichmentEntry]
    status: MoonEnrichmentStatus
    cached_count: int
    pending_dates: list[str] = Field(default_factory=list)

File: frontend/src/types/moon-enrichment.ts
export type MoonEnrichmentStatus = 'complete' | 'partial' | 'pending' | 'unavailable'

export interface MoonEnrichmentEntry {
  date: string
  phase_name: string
  illumination_pct: number
  age_days?: number | null
  is_waxing?: boolean | null
  special_labels: string[]
  visual_url?: string | null
}

export interface MoonEnrichmentResponse {
  entries: MoonEnrichmentEntry[]
  status: MoonEnrichmentStatus
  cached_count: number
  pending_dates: string[]
}

export interface MoonEnrichmentByDate {
  [date: string]: MoonEnrichmentEntry
}
